dojo_plugin/utils/__init__.py

```diff
@@ -169,18 +169,19 @@ def resolved_tar(dir, *, root_dir, filter=None):
 
 def module_visible(dojo, module, user):
     return (
-        "time_visible" not in module or
-        module["time_visible"] <= datetime.datetime.now(pytz.utc) or
+        "visibility" not in module or
+        module["visibility"]["start"] <= datetime.datetime.now(pytz.utc) or
         is_dojo_admin(user, dojo)
     )
 
 
 def module_challenges_visible(dojo, module, user):
-    return (
-        "time_assigned" not in module or
-        module["time_assigned"] <= datetime.datetime.now(pytz.utc) or
-        is_dojo_admin(user, dojo)
-    )
+     return (
+         ("time_assigned" not in module and "visibility" not in module) or
+         ("time_assigned" in module and module["time_assigned"] <= datetime.datetime.now(pytz.utc)) or
+         ("visibility" in module and module["visibility"] <= datetime.datetime.now(pytz.utc)) or
+         is_dojo_admin(user, dojo)
+     )
```

Comment:
Can we delete these two methods? They look like dead code to me. I think they are defined, imported, and then never used.

I know this is orthogonal, but it's better than refactoring dead code!

Reply:
I don't believe we can delete these.

I had the exact same thought and initially did exactly that initially.  The `module_challenge_visible` is used in the course template to determine which modules will be included in grading.

The concept of a module's "time_assigned" must be distinct from the module visibility if we intend on supporting the use-case where a module is visible to students, but not yet "assigned" for grading purposes.  While I do not personally believe I will ever use it, someone could.  Additionally, removing it has the potential of breaking any dojos that currently rely the `time_assigned` field.  I could certainly rename one for a better read.

Reply:
```
$ git grep 'module_visible'
dojo_plugin/pages/course.py:from ..utils import module_visible, module_challenges_visible, is_dojo_admin
dojo_plugin/utils/__init__.py:def module_visible(dojo, module, user):
$ git grep 'module_challenges_visible'
dojo_plugin/pages/course.py:from ..utils import module_visible, module_challenges_visible, is_dojo_admin
dojo_plugin/utils/__init__.py:def module_challenges_visible(dojo, module, user):
```

Am I confused? Where is it? I think `time_assigned` is a dead concept.

Reply:
You are correct.

I'm going claim this is evidence that sleep does in fact have value.